linetools/spectra/xspectrum1d.py

```diff
@@ -84,92 +75,205 @@ def from_file(self, ifile, **kwargs):
           Filename
         """
         # put import here to avoid circular import with io.py
-        from .io import readspec
-        slf = readspec(ifile, **kwargs)
+        #from .io import readspec
+        from linetools.spectra import io as tio
+        reload(tio)
+        slf = tio.readspec(ifile, **kwargs)
         return slf
 
     @classmethod
     def from_spec1d(cls, spec1d):
         """ Input Spectrum1D
-        """
-        # Giddy up
-        slf = cls(flux=spec1d.flux, wcs=spec1d.wcs, unit=spec1d.unit,
-                  uncertainty=spec1d.uncertainty, mask=spec1d.mask,
-                  meta=spec1d.meta.copy())
-        slf.co = (spec1d.co if hasattr(spec1d, 'co') else None)
 
+        Avoids error array for now
+        """
+        slf = cls.from_tuple((spec1d.dispersion, spec1d.flux))
+        #slf = cls(flux=spec1d.flux, wcs=spec1d.wcs, unit=spec1d.unit,
+        #          uncertainty=spec1d.uncertainty, mask=spec1d.mask,
+        #          meta=spec1d.meta.copy())
```

Comment:
Ok, could you add to the docstring that it should take a specutils.Spectrum1D as input?